### PR TITLE
Added named export `proj4` support to index.js

### LINKED ISSUE #476: ESM vs CommonJS exports inconsistency
By default, the `dist/proj4-src.js` file generates an 'exports' field using the grunt build, which does not provide a default export,

meanwhile, the module `lib/index.js` provides only a default export.

In environments where there is a mismatch between library usage (for example, we experience this when building our app as ESM, which uses the `lib/index.js`, but test using jest which only supports commonjs transformation and as such enters the `dist/proj4-src.js` file) the simple:
```ts
import proj4 from 'proj4';
```

breaks, as the tests which use commonjs transformation use the dist file, which throws the error that proj4 is undefined (there is no default export).

the solution is to support the named export proj4, such that both:

```ts
import * as proj4 from 'proj4';
```

will work with both `module` and `main` entry points.

### PR DESCRIPTION
**Pull request by Beilinson**
Fixes #476 

I hope this can be merged soon, as this issue is actively breaking unit tests in our GIS projects, and additionally is good practice to not have only default exports

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "proj4",
-  "version": "2.12.2-alpha",
+  "version": "2.12.3-alpha",
```
Comment (ahocevar) [inline package.json:3]: Please remove this change - that will be done as part of release management.

**Review by ahocevar — CHANGES_REQUESTED** [review]

**Review by ahocevar — APPROVED** [review]

**Review by ahocevar — CHANGES_REQUESTED** [review]
As you can see, this breaks the build.

**Comment by Beilinson** [comment]
Yep, looking into it locally

**Comment by Beilinson** [comment]
From my local tests this works, and also manages to fix the problem with jest in a different way, by exposing shimming the .default property that some parsers use when transforming ESM code to CJS.

I noticed that this is the same change as from #263 which is 7 years old, any reason that wasn't ever merged?

**Comment by ahocevar** [comment]
It's really about time to update the build system - along the lines of #462, but getting rid of grunt entirely.

In the meantime, I think a safer change would be to create a separate entry point file for rollup (e.g. `./rollup/index.js`), which imports `proj4` from `../lib/index.js` and applies the hack you're proposing. Then you could point rollup to `rollup/index.js` instead of `lib/index.js` in the `Gruntfile`.

Then when updating the build system, we just get rid of `rollup/index.js`, and everything should work without this hack.

**Comment by Beilinson** [comment]
having done some more tests, I realized that what this PR achieves can also be done by enabling [esModuleInterop](https://www.typescriptlang.org/tsconfig/esModuleInterop.html) which does exactly what this fixes ascribes. Closing this, if you would like any help with continuing #462 (for example, upgrading to jest as well, which supercedes mocha) I would be glad to help

**Comment by ahocevar** [comment]
@Beilinson I'd prefer vitest over jest, but help is always appreciated. Thanks!

**Pull request closed without merging**